### PR TITLE
Issue 43 page title provider

### PR DESCRIPTION
**Pull request by format-gmbh**
add class BlogPageTitleProvider for use in the PostController
outputs the title of the blog in the indexAction and the title of the post in the showAction

Releases: main

### CODE REVIEW

File: Classes/PageTitle/BlogPageTitleProvider.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ *
+ * The TYPO3 project - inspiring people to share!
+ */
+
+/**
+ * page title provider class for blog post index and detail view
+ */
+final class BlogPageTitleProvider extends AbstractPageTitleProvider
+{
+    protected string $title = '';
```
Comment (brotkrueml) [inline Classes/PageTitle/BlogPageTitleProvider.php:27]: Please remove the definition of the title completely:

PHP Fatal error:  Type of T3docs\BlogExample\PageTitle\BlogPageTitleProvider::$title must not be defined

File: Configuration/TypoScript/setup.typoscript [hunk 2]
```diff
@@ -50,3 +50,12 @@ module.tx_blogexample {
 
 lib.blog_example_post_list < tt_content.list.20.blogexample_postlist
 
+# Configure page title provider for page title in post detail view
```
Comment (brotkrueml) [inline Configuration/TypoScript/setup.typoscript:53]: "post detail view" is not really accurate?

**Review by brotkrueml — CHANGES_REQUESTED** [review]
Thanks for the improvement, I have some remarks ;-)

Additionally, what do you think, should we backport this to 12.4?

File: Classes/PageTitle/BlogPageTitleProvider.php (hunk 1 above)
Comment (format-gmbh) [reply Classes/PageTitle/BlogPageTitleProvider.php]: done

File: Configuration/TypoScript/setup.typoscript (hunk 2 above)
Comment (format-gmbh) [reply Configuration/TypoScript/setup.typoscript]: done

**Review by brotkrueml — APPROVED** [review]
@format-gmbh Thanks for the improvement :+1:

**Pull request merged**